src/components/V2ExchangeRedirectModal/index.tsx

```diff
@@ -0,0 +1,90 @@
+import React, { useState, useEffect } from 'react'
+import styled from 'styled-components'
+import { Modal, Text, Link, Button, Flex, Heading } from '@pancakeswap-libs/uikit'
+
+const defaultOnDismiss = () => null
+
+const StyledLink = styled(Link)`
+  width: 100%;
+`
+
+const Divider = styled.div`
+  margin: 24px 0 12px;
+  width: 100%;
+  height: 1px;
+  background-color: ${({ theme }) => theme.colors.borderColor};
+`
+
+type V2ExchangeRedirectModalProps = {
+  onDismiss?: () => void
+  handleCloseModal: () => void
+}
+
+const V2ExchangeRedirectModal = ({ onDismiss = defaultOnDismiss, handleCloseModal }: V2ExchangeRedirectModalProps) => {
+  const [timerSecondsRemaining, setTimerSecondsRemaining] = useState(5)
+
+  useEffect(() => {
+    const tick = () => {
+      setTimerSecondsRemaining((prevSeconds) => prevSeconds - 1)
+    }
+
+    const timerInterval = setInterval(() => tick(), 1000)
+
+    const preventClickHandler = (e) => {
```

Comment:
Preventing clicks as with the other recently implemented modal in the repo. It can be closed at any time via the close button though.